Task3_Yoav_Uzan/src/Board.cpp

```diff
@@ -279,120 +283,86 @@ void Board::replaceTurn()
 //check if the king is in danger
 bool Board::checkIfKingInDanger(Loction KingLoc, Color kingColor, vector<vector<Peice*>> tempBoard)

```

Comment:
your code is complex and you have a lot of unnecessary functions, and a lot of loops